src/generic-container/generic-container.test.ts

```diff
@@ -368,6 +371,21 @@ describe("GenericContainer", () => {
     await container.stop();
   });
 
+  it("should stream logs from a running container after restart", async () => {
+    const mock = jest.mocked(containerLog);
```

Comment:
There is already a `FakeLogger` implementation in `logger.ts`, is it possible to use that?

Reply:
I wanted to use it but could not see a way mock it out or replace it at test time, if you have any ideas I am happy to implement them.

Reply:
Yes actually it's not straight forward. There's a mix of fakes+DI and jest mocking, it would be good to pick one at some point, but it is outside the scope of this PR

Reply:
I feel that a good approach might be providing a `Logger` as an parameter to `GenericContainer` which can be propagated elsewhere, it could have a default value of the `logger.containerLog` but then be overridden in tests (or by users who want to override it), but this would be best as a seperate PR. Would be happy to take it further if you think the approach is sound.

Reply:
It's a good idea, I don't know at the moment what's the right thing to do. I quite like that right now you can call functions like `containerLogs(container)` from anywhere without worrying about dependencies, and so I don't love the idea of propagating out its dependencies. 

In fact, I wonder if in this case our level of testing is too low, we're testing the actual `containerLogs` function here, instead of testing that we call it when a container is restarted. Perhaps then we should just spy on `containerLogs` and ensure it's called when the container is restarted (instead of mocking the logger, checking the actual logs, etc)?

Reply:
Let me spend a little more time trying to get a spy working, it did not when I initially tried.